components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/exception/ApplicationRolesException.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authentication.framework.exception;
+
+import org.wso2.carbon.identity.base.IdentityException;
+
+/**
+ * Application roles exception.
+ */
+public class ApplicationRolesException extends IdentityException {
```

Comment:
Rename to adhere

Reply:
Addressed in 70aa404975af17179c9939c5dc4f7448f2e05423
Kept the exception name as it is and added server and client exceptions and error codes.
Error codes are currently in the identity-framework.
Should these be moved to where actual implementation is ?